### PR TITLE
fix sqldb parameter validator

### PR DESCRIPTION
**Pull request by zhongyi-zhang**

### CODE REVIEW

File: test/unit/services/azuresqldb/provision-spec.js [hunk 1]
```diff
@@ -135,7 +135,7 @@ describe('SqlDb - Provision - PreConditions', function () {
         });
 
         it('should fail to validate the parameters', function (done) {
```
Comment (niroyb) [inline test/unit/services/azuresqldb/provision-spec.js]: This test is not async. No need to have done()

File: test/unit/services/azuresqldb/provision-spec.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ describe('SqlDb - Provision - PreConditions', function () {
...
-            (cp.getInvalidParams().length).should.not.equal(0);
```
Comment (niroyb) [inline test/unit/services/azuresqldb/provision-spec.js]: Maybe also validate which parameters are invalid

File: test/unit/services/azuresqldb/provision-spec.js (hunk 1 above)
Comment (niroyb) [inline test/unit/services/azuresqldb/provision-spec.js]: "should find invalid parameters"

File: test/unit/services/azuresqldb/provision-spec.js [hunk 3]
```diff
@@ -134,9 +134,9 @@ describe('SqlDb - Provision - PreConditions', function () {
             cp = new cmdProvision(params);
         });
 
-        it('should fail to validate the parameters', function (done) {
-            (cp.getInvalidParams().length).should.not.equal(0);
-            done();
+        it('should find invalid parameters', function () {
+            (cp.getInvalidParams().length).should.equal(7);
+            (cp.getInvalidParams().toString()).should.equal('resourceGroupName,location,sqlServerName,sqldbName,administratorLogin,administratorLoginPassword,collation');
```
Comment (niroyb) [inline test/unit/services/azuresqldb/provision-spec.js]: use should.deepEqual() instead of converting to string

Comment (zhongyi-zhang) [inline test/unit/services/azuresqldb/provision-spec.js]: @niroyb Thanks! Fixed it.

**Pull request merged**